packages/discovery-provider/plugins/pedalboard/apps/solana-relay/src/routes/relay/shouldAttachGeoData.test.ts

```diff
@@ -0,0 +1,64 @@
+import { PaymentRouterProgram } from '@audius/spl'
+import {
+  Keypair,
+  PublicKey,
+  TransactionInstruction,
+} from '@solana/web3.js'
+import { config } from '../../config'
+import { describe, it } from 'vitest'
+import { shouldAttachGeoData } from './shouldAttachGeoData'
+import assert from 'assert'
+
+const PAYMENT_ROUTER_PROGRAM_ID = new PublicKey(config.paymentRouterProgramId)
+const MEMO_V2_PROGRAM_ID = new PublicKey(
+  'MemoSq4gqABAXKb96qnH8TysNcWxMyWCqXgDLGmfcHr'
+)
+
+const getRandomPublicKey = () => Keypair.generate().publicKey
+
+describe('Should Attach Geo Data', function () {
+    it('Attach when instructions are a payment', async function () {
+      const [pda, bump] = PublicKey.findProgramAddressSync(
+        [new TextEncoder().encode('payment_router')],
+        new PublicKey(PAYMENT_ROUTER_PROGRAM_ID)
+      )
+      const instructions = [
+        await PaymentRouterProgram.createRouteInstruction({
+          sender: getRandomPublicKey(),
+          senderOwner: pda,
+          paymentRouterPdaBump: bump,
+          recipients: [getRandomPublicKey()],
+          amounts: [BigInt(100)],
+          totalAmount: BigInt(100),
+          programId: PAYMENT_ROUTER_PROGRAM_ID
+        })
+      ]
+      const attach = await shouldAttachGeoData(instructions)
```

Comment:
Is there any way we could test the transaction itself and make sure the instruction data is correct?

Reply:
good looks! updated the code to make this easier and added a test